src/main/java/mate/academy/bookshop/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,63 @@
+package mate.academy.bookshop.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookshop.dto.shoppingcart.CartItemRequestDto;
+import mate.academy.bookshop.dto.shoppingcart.CartResponseDto;
+import mate.academy.bookshop.dto.shoppingcart.UpdateCartItemRequestDto;
+import mate.academy.bookshop.exception.EntityNotFoundException;
+import mate.academy.bookshop.mapper.CartItemMapper;
+import mate.academy.bookshop.mapper.ShoppingCartMapper;
+import mate.academy.bookshop.model.CartItem;
+import mate.academy.bookshop.model.ShoppingCart;
+import mate.academy.bookshop.repository.CartItemRepository;
+import mate.academy.bookshop.repository.ShoppingCartRepository;
+import mate.academy.bookshop.repository.UserRepository;
+import mate.academy.bookshop.service.ShoppingCartService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final UserRepository userRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    public CartResponseDto getShoppingCart(Long userId) {
+        ShoppingCart shoppingCart = shoppingCartRepository.findByUserId(userId);
```

Comment:
What if shopping does not exists?

Reply:
I create shopping cart after user being registered, so if user authenticated -> shopping cart must be present for this user.

Reply:
it's ok I guess but would still be nice to add checks, just in case something will change in the future, up to you